sw/device/lib/runtime/epmp.h

```diff
@@ -0,0 +1,550 @@
+ *
+ * @param entry Entry index to update (0 <= `entry` < `kEpmpNumRegions`)
+ * @param region Region start and end addresses. Must be 4 byte aligned.
+ * @param permissions Updated permissions to write to pmpcfg for `entry`.
+ * @return `epmp_entry_configure_result_t`.
+ */
+epmp_entry_configure_result_t epmp_entry_configure_na4(epmp_entry_t entry,
+                                                       epmp_region_t region,
+                                                       epmp_perm_t permissions);
+
+/**
+ * Configures EPMP address matching to Naturally Aligned Power-Of-Two (NAPOT).
+ *
+ * Address matching is configured as NAPOT for `entry`. The length of `region`
+ * must be a power of two greater than four. The `region` (both start and end
+ * addresses) must also be aligned to the same power of two.
+ *
+ * If `kEpmpGranularity` is greater than zero then the entire `region` must
+ * also be aligned to `2 ** (2 + kEpmpGranularity)`.
+ *
+ * IMPORTANT: the entry will not be updated until `epmp_transaction_end`
+ * is called.
+ *
+ * Example:
+ *
+ *   ...
+ *   res0 = epmp_transaction_start(kEpmpEntryCount2);
+ *   res1 = epmp_entry_configure_napot(kEpmpEntry0,
+ *              (epmp_region_t){ .start = 0x10, .end = 0x20 },
+ *              kEpmpPermLockedReadOnly);
+ *   res2 = epmp_entry_configure_napot(kEpmpEntry1,
+ *              (epmp_region_t){ .start = 0x50, .end = 0x58 },
+ *              kEpmpPermLockedReadWrite);
+ *   res3 = epmp_transaction_end(kEpmpEntryCount2);
+ *   ...
+ *
+ * Result:
+ *
+ *   Entry | Value of `pmpaddr`        | Value of `pmpcfg` |
+ *   ======+===========================+===================+
+ *       0 | 0x41 ((0x10 >> 2) | 0b01) |         0b1011001 |
+ *       1 | 0x14 ((0x50 >> 2) | 0b00) |         0b1011011 |
+ *
+ * @param entry Entry index to update (0 <= `entry` < `kEpmpNumRegions`)
+ * @param region Region start and end addresses.
+ * @param permissions Updated permissions to write to pmpcfg for `entry`.
+ * @return `epmp_entry_configure_result_t`.
+ */
+epmp_entry_configure_result_t epmp_entry_configure_napot(
+    epmp_entry_t entry, epmp_region_t region, epmp_perm_t permissions);
+
+/**
+ * Toggle for Rule Locking Bypass (mseccfg.RLB).
+ */
+typedef enum {
+  /**
+   * Rule Locking Bypass disabled (mseccfg.RLB = 0).
+   */
+  kEpmpRlbToggleDisabled,
+
+  /**
+   * Rule Locking Bypass enabled (mseccfg.RLB = 1).
+   */
+  kEpmpRlbToggleEnabled,
+} epmp_rlb_toggle_t;
+
+/**
+ * Enable or disable the Rule Locking Bypass (RLB) feature.
+ *
+ * When enabled (mseccfg.RLB = 1) the Rule Locking Bypass features allows
+ * locked entries to be modified. If any PMP entries are locked and RLB
+ * is disabled (mseccfg.RLB = 0) then it is no longer possible to enable
+ * RLB. RLB will be set as requested or an error will be returned.
+ *
+ * @param toggle Whether RLB should be enabled or disabled.
+ * @return `epmp_result_t`
+ */
+epmp_result_t epmp_set_rule_locking_bypass(epmp_rlb_toggle_t toggle);
```

Comment:
Already mentioned in a comment above, but WDYT about making this clear-only and removing the getter below? Also, is this a part of a transaction?

Reply:
I think clear only makes sense. If we're going to define any entries in the initial assembly then we'll need to enable RLB at the same time.

Clearing the RLB would not be part of a transaction. I should probably also mandate that a transaction is not in progress when this is performed. I don't think that updating RLB as part of a transaction has much value, but we could consider it.